libfoundation/include/foundation.h

```diff
@@ -1244,11 +1290,39 @@ template<typename T> inline bool MCMemoryResizeArray(uindex_t p_new_count, T*& x
 	return false;
 }
 
+template <typename T>
+bool MCMemoryResizeArrayInit(uindex_t p_new_count, T*& x_array, uindex_t& x_count)
+{
+    // Capture the current count before resizing
+    uindex_t t_current_count = x_count;
+    if (MCMemoryResizeArray(p_new_count, x_array, x_count))
+    {
+        // Default construct any new elements that were allocated
+        for (uindex_t i = t_current_count; i < p_new_count; i++)
+            new (&x_array[i]) T();
+        return true;
+    }
+    
+    return false;
+}
+
 template<typename T> void MCMemoryDeleteArray(T* p_array)
 {
 	MCMemoryDeleteArray(static_cast<void *>(p_array));
 }
 
+// Array deleter that runs the destructor for each element
+template <typename T>
+void MCMemoryDeleteArray(T* p_array, size_t N)
```

Comment:
Here you've used `size_t` for the count.  This is correct, but everything else currently uses `uindex_t` for the count, so it might be a good idea to be consistent.